packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useFetchMappingExtractionStatus.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { QueryClient } from "@tanstack/react-query";
+import { useQuery } from "@tanstack/react-query";
+import type { IExtractionClient } from "@itwin/insights-client";
+import { ExtractorState } from "@itwin/insights-client";
+import type { Mapping } from "@itwin/insights-client";
+import { STATUS_CHECK_INTERVAL } from "../../Constants";
+import { ExtractionStates } from "../Extraction/ExtractionStatus";
+import type { GetAccessTokenFn } from "../../context/GroupingApiConfigContext";
+import { useExtractionClient } from "../../context/ExtractionClientContext";
+import { useMemo } from "react";
+import { useExtractionStateJobContext } from "../../context/ExtractionStateJobContext";
+
+export interface MappingExtractionStatusProps {
+  getAccessToken: GetAccessTokenFn;
+  mapping: Mapping;
+  enabled: boolean;
+}
+
+export interface MappingQueryResults {
+  mappingId: string;
+  finalExtractionStateValue: ExtractionStates;
+}[];
+
+export const fetchMappingStatus = async (
+  mappingId: string,
+  jobId: string,
+  getAccessToken: GetAccessTokenFn,
+  extractionClient: IExtractionClient,
+) => {
+  const accessToken = await getAccessToken();
+
+  if(jobId === ""){
+    return { mappingId, finalExtractionStateValue: ExtractionStates.None };
+  }
+  const extractionStatusResponse = await extractionClient.getExtractionStatus(accessToken, jobId);
+
+  switch(extractionStatusResponse.state) {
+    case undefined:
+      return { mappingId, finalExtractionStateValue: ExtractionStates.Starting};
+    case ExtractorState.Running:
+      return { mappingId, finalExtractionStateValue: ExtractionStates.Running};
+    case ExtractorState.Failed:
+      return { mappingId, finalExtractionStateValue: ExtractionStates.Failed};
+    case ExtractorState.Queued:
+      return { mappingId, finalExtractionStateValue: ExtractionStates.Queued};
+    case ExtractorState.Succeeded:
+      return { mappingId, finalExtractionStateValue: ExtractionStates.Succeeded};
+    default:
+      return { mappingId, finalExtractionStateValue: ExtractionStates.None};
+  }
+};
+
+export const createQueryForEachJobId = (mappingId: string, jobId: string, getAccessToken: GetAccessTokenFn, extractionClient: IExtractionClient, enabled: boolean) => ({
```

Comment:
How can it create something "for each", when only 1 thing is given?

Reply:
I have changed the method's name.